### PR TITLE
docker_network: support ipv6

### PR DESCRIPTION
**Pull request by rmoriz**
### Description

allow setting the [EnableIPv6](https://docs.docker.com/engine/api/v1.30/#operation/NetworkCreate) option to create networks with IPv6 support.

### CODE REVIEW

**Comment by rmoriz** [comment]
issue #876

test failure caused by unrelated change (Fedora: No candidate version available for docker-engine) :(

**Comment by rmoriz** [comment]
(maybe) blocked by issue in #890 - I think there should be no requirement to specify a gateway address but currently it does PR does not work without setting one 😒

File: test/cookbooks/docker_test/recipes/network.rb [hunk 1]
```diff
@@ -225,3 +225,19 @@
   network_name 'network_h1'
   action :disconnect
 end
+
+###########
+# network_i
+###########
+
+# IPv6 enabled network
+docker_network 'network_i1' do
```
Comment (chasebolt) [inline test/cookbooks/docker_test/recipes/network.rb]: rename to `network_ipv6`

File: test/cookbooks/docker_test/recipes/network.rb [hunk 2]
```diff
@@ -225,3 +225,19 @@
...
+  enable_ipv6 true
+  subnet 'fd00:dead:beef::/48'
+  action :create
+end
+
+# IPv6 disabled network (default)
+docker_network 'network_i2' do
```
Comment (chasebolt) [inline test/cookbooks/docker_test/recipes/network.rb]: rename to `network_ipv4`

File: test/cookbooks/docker_test/recipes/network.rb (hunk 1 above)
Comment (rmoriz) [inline test/cookbooks/docker_test/recipes/network.rb]: My intent was to keep the naming consistant with the Docker API https://docs.docker.com/engine/api/v1.30/#operation/NetworkCreate - the docker network CLI option is "--ipv6".

**Comment by rmoriz** [comment]
Okay, let me also add the "internal" network flag

File: libraries/helpers_network.rb [hunk 3]
```diff
@@ -92,7 +92,6 @@ def consolidate_ipam(subnets, ranges, gateways, auxaddrs)
             data[s]['Gateway'] = g
             match = true
           end
-          raise "no matching subnet for gateway #{s}" unless match
```
Comment (chasebolt) [inline libraries/helpers_network.rb]: is there a gateway being set? still confused on how this is being triggered if `gateways` is empty.

Comment (rmoriz) [inline libraries/helpers_network.rb]: the inspect looks like:

```json

 "IPAM": {
            "Driver": "default",
            "Options": null,
            "Config": [
                {
                    "Subnet": "172.24.0.0/16",
                    "Gateway": "172.24.0.1"
                },
                {
                    "Subnet": "fd00:dead:beef::/48"
                }
            ]
        },
```
Docker automatically adds the IPv4 part (Gateway set) but the IPv6 subnet never matches a Gateway.

Comment (chasebolt) [inline libraries/helpers_network.rb]: good enough for me. thanks for the clarification.

**Comment by rmoriz** [comment]
sadly this still has mayor issues with idempotency. And I must say this type of library-cookbook is super hard to understand :(

**Pull request merged**